src/server/game/Entities/Player/PlayerStorage.cpp

```diff
@@ -5499,6 +5499,22 @@ bool Player::LoadFromDB(ObjectGuid playerGuid, CharacterDatabaseQueryHolder cons
                     SetAcceptWhispers(true);
                 break;
         }
+
+        switch (sWorld->getIntConfig(CONFIG_GM_MAIL_TO))
+        {
+            default:
```

Comment:
never see default in the first line of a switch

Reply:
Looks like it has to be there because the default is to run case 0

Reply:
It's used the same for the 4 switches above, so I just copied it

Reply:
This may seem off-topic but all those switch statements accumulate to 75 lines of code which can be shorten down to 17 lines

```c++

// GM state
const auto ApplyExtraFlags = [&](WorldIntConfigs configIndex, uint32 flags, void (Player::*func)(bool), bool val)
{
    if (!AccountMgr::IsPlayerAccount(GetSession()->GetSecurity()))
    {
        switch (sWorld->getIntConfig(configIndex))
        {
            case 1: (this->*func)(val); break;
            case 2: if (extraflags & flags) (this->*func)(val); break;
        }
    }
};

ApplyExtraFlags(CONFIG_GM_CHAT,          PLAYER_EXTRA_GM_CHAT,         SetGMChat,         true);
ApplyExtraFlags(CONFIG_GM_MAIL_TO,       PLAYER_EXTRA_ACCEPT_MAIL,     SetAcceptMail,     true);
ApplyExtraFlags(CONFIG_GM_INVITES_TO,    PLAYER_EXTRA_ACCEPT_INVITES,  SetInvites,        true);
ApplyExtraFlags(CONFIG_GM_LOGIN_STATE,   PLAYER_EXTRA_GM_ON,           SetGameMaster,     true);
ApplyExtraFlags(CONFIG_GM_VISIBLE_STATE, PLAYER_EXTRA_GM_INVISIBLE,    SetGMVisible,      false);
```

Thoughts?

Reply:
I like this, but remember that if we do this, now we have to test every if statement we modified to be sure we didn't break something. May not be worth it

Reply:
But actually it might not be so bad either, given that the code will be easier to maintain in the future

Reply:
I will not change the logic in this PR. This is better for a refactor in another pr